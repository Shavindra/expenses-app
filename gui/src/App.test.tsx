/**
 * External Imports
 */
import initStoryshots from '@storybook/addon-storyshots';

// Run story snapshots for entire app
// CRA doesn't allow "root" property change. So include it here.
// ref: https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/scripts/utils/createJestConfig.js#L73
initStoryshots();
