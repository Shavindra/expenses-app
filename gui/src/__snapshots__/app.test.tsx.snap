// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots App Bar Closed State 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionAbsolute MuiAppBar-colorPrimary AppBarCmp-appBar-371 MuiPaper-elevation0"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular AppBarCmp-toolbar-369 MuiToolbar-gutters"
  >
    <button
      aria-label="open drawer"
      className="MuiButtonBase-root MuiIconButton-root AppBarCmp-menuButton-373 MuiIconButton-colorInherit MuiIconButton-edgeStart"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
    </button>
    <h1
      className="MuiTypography-root AppBarCmp-title-375 MuiTypography-h6 MuiTypography-noWrap"
    >
      Dashboard
    </h1>
  </div>
</header>
`;

exports[`Storyshots App Bar Open State 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionAbsolute MuiAppBar-colorPrimary AppBarCmp-appBar-268 AppBarCmp-appBarShift-269 MuiPaper-elevation0"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular AppBarCmp-toolbar-266 MuiToolbar-gutters"
  >
    <button
      aria-label="open drawer"
      className="MuiButtonBase-root MuiIconButton-root AppBarCmp-menuButton-270 AppBarCmp-menuButtonHidden-271 MuiIconButton-colorInherit MuiIconButton-edgeStart"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
    </button>
    <h1
      className="MuiTypography-root AppBarCmp-title-272 MuiTypography-h6 MuiTypography-noWrap"
    >
      Dashboard
    </h1>
  </div>
</header>
`;

exports[`Storyshots App Default 1`] = `
<div
  className="App"
>
  <div
    className="page"
  >
    <header
      className="MuiPaper-root-25 MuiAppBar-root-14 MuiAppBar-positionAbsolute-16 MuiAppBar-colorPrimary-21 AppBarCmp-appBar-8 MuiPaper-elevation0-28"
    >
      <div
        className="MuiToolbar-root-53 MuiToolbar-regular-55 AppBarCmp-toolbar-6 MuiToolbar-gutters-54"
      >
        <button
          aria-label="open drawer"
          className="MuiButtonBase-root-66 MuiIconButton-root-57 AppBarCmp-menuButton-10 MuiIconButton-colorInherit-60 MuiIconButton-edgeStart-58"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label-65"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-69"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
        <h1
          className="MuiTypography-root-78 AppBarCmp-title-12 MuiTypography-h6-88 MuiTypography-noWrap-97"
        >
          Dashboard
        </h1>
      </div>
    </header>
    <div
      className="MuiDrawer-root-111 MuiDrawer-docked-112"
    >
      <div
        className="MuiPaper-root-25 MuiDrawer-paper-113 DrawerCmp-drawerPaper-108 DrawerCmp-drawerPaperClose-109 MuiDrawer-paperAnchorLeft-114 MuiDrawer-paperAnchorDockedLeft-118 MuiPaper-elevation0-28"
      >
        <div
          className="DrawerCmp-toolbarIcon-110"
        >
          <button
            className="MuiButtonBase-root-66 MuiIconButton-root-57"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label-65"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-69"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </span>
          </button>
        </div>
        <hr
          className="MuiDivider-root-123"
        />
        <ul
          className="MuiList-root-130 MuiList-padding-131"
        >
          <a
            aria-current="page"
            aria-disabled={false}
            className="MuiButtonBase-root-66 MuiListItem-root-134 MuiListItem-gutters-141 MuiListItem-button-142 Mui-selected"
            href="/"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            style={Object {}}
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root-145"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-69"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root-147"
            >
              <span
                className="MuiTypography-root-78 MuiListItemText-primary-151 MuiTypography-body1-80 MuiTypography-displayBlock-107"
              >
                Dashboard
              </span>
            </div>
          </a>
          <a
            aria-current={null}
            aria-disabled={false}
            className="MuiButtonBase-root-66 MuiListItem-root-134 MuiListItem-gutters-141 MuiListItem-button-142"
            href="/expenses"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemIcon-root-145"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-69"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 18v1c0 1.1-.9 2-2 2H5c-1.11 0-2-.9-2-2V5c0-1.1.89-2 2-2h14c1.1 0 2 .9 2 2v1h-9c-1.11 0-2 .9-2 2v8c0 1.1.89 2 2 2h9zm-9-2h10V8H12v8zm4-2.5c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root-147"
            >
              <span
                className="MuiTypography-root-78 MuiListItemText-primary-151 MuiTypography-body1-80 MuiTypography-displayBlock-107"
              >
                Expenses
              </span>
            </div>
          </a>
        </ul>
      </div>
    </div>
    <main
      className="LayoutMainCmp-content-3"
    >
      <div
        className="LayoutMainCmp-appBarSpacer-2"
      />
      <div
        className="MuiContainer-root-153 LayoutMainCmp-container-4 MuiContainer-maxWidthLg-159"
      >
        <div
          className="MuiGrid-root-161 MuiGrid-container-162 MuiGrid-spacing-xs-3-186"
        >
          <div
            className="MuiGrid-root-161 MuiGrid-item-163 MuiGrid-grid-xs-12-207"
          >
            Home Page
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`Storyshots Text Fields Basic Text Field 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
</div>
`;

exports[`Storyshots Text Fields Editable Text Field 1`] = `
<div
  className="makeStyles-root-338"
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled"
        disabled={true}
        id="ediable-field"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <span
    aria-describedby={null}
    aria-label="edit"
    className=""
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title="edit"
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </span>
</div>
`;

exports[`Storyshots Text Fields Full Text Field 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
</div>
`;
